uint8[] row0
uint8[] row1
uint8[] row2
uint8[] row3
uint8[] row4
uint8[] row5
uint8 player
uint8 report